packages/react-scripts/scripts/eject.js

```diff
@@ -36,6 +37,31 @@ prompt(
     process.exit(1);
   }
 
+  // Make sure there are no dirty git status
+  function statusSync() {
+    try {
+      let stdout = execSync(`git status --porcelain`).toString();
+      let status = { dirty: 0, untracked: 0 };
```

Comment:
You are right, I think untracked files should be exclude. I will make pull request on this tonight.